### DISCUSSION TITLE
Fonts variables (in TailwindCSS config) not applying correctly in Next.js

### DISCUSSION THREAD

**Discussion by kimgiutae** [tailwindlabs/tailwindcss#12910] [Help] [answered]
### Summary

I'm following the guide provided in the next.js docs for loading google fonts using TailwindCSS: https://nextjs.org/docs/app/building-your-application/optimizing/fonts#with-tailwind-css.

I imported 2 fonts, open sans for general font and raleway for titles.

```javascript
import { Raleway, Open_Sans } from 'next/font/google'

const open = Open_Sans({
  weight: ['400', '700'],
  style: ['normal', 'italic'],
  subsets: ['latin'],
  display: 'swap',
  variable: '--font-open'
})

const raleway = Raleway({
  weight: ['300', '600', '900'],
  style: ['normal'],
  subsets: ['latin'],
  display: 'swap',
  variable: '--font-raleway'
})
```

I also added the css variables to the TailwindCSS config file:

```javascript
module.exports = {
  content: ['./src/components/*.tsx', './src/app/**/*.tsx'],
  theme: {
    extend: {
      fontFamily: {
        open: ['var(--font-open)'],
        raleway: ['var(--font-raleway)']
      }
    }
  }
}
```

But when I add the classnames to the `Layout` component:

```javascript
const RootLayout = ({ children }) => {
  return (
    <html lang='en' className={`font-open ${open.className} ${raleway.className}`}>
      <body>
        <main>
          {children}
          <h1 className='font-raleway'>Title</h1>
        </main>
      </body>
    </html>
  )
}
```

The raleway font overrides the open sans font. **No matter if I use the class helper "font-open" or "font-raleway"**.

The only way that seems to function is using the `open.className` or `raleway.className` string directly in the className of the desired component, in example: `Title`.

I'm not shure if the problem is in next.js or TailwindCSS, just following the docs.

**My guess is that the raleway css props are setted later in the bundle generated by TailwindCSS and then overriding the open sans font. But in that case, I'm doing smth wrong?**

How can I use multiple fonts in the next.js app with TailwindCSS.

### Additional information

next: 14.1.0
tailwindcss: 3.4.1

**Comment by makibat** [comment, marked as answer]
I think you're supposed to use the font's `CSS variable`, rather than its `className`:

``` js
const RootLayout = ({ children }) => {
  return (
    <html lang='en' className={`font-open ${open.variable} ${raleway.variable}`}>
      <body>
        <main>
          {children}
          <h1 className='font-raleway'>Title</h1>
        </main>
      </body>
    </html>
  )
}
```

Now you should be able to use your `font-open` and `font-raleway` classes without any overriding issues.

**Reply by kimgiutae** [reply]
That was the problem. Thanks @makibat. Sometimes the issue is simplier we imagine. 🤣
